api/server/handlers/porter_app/create_secret_and_open_pr.go

```diff
@@ -59,12 +66,16 @@ func (c *OpenStackPRHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		// generate porter jwt token
 		jwt, err := token.GetTokenForAPI(user.ID, project.ID)
 		if err != nil {
-			c.HandleAPIError(w, r, apierrors.NewErrInternal(fmt.Errorf("error getting token for API: %w", err)))
+			err = fmt.Errorf("error getting token for API: %w", err)
+			err := telemetry.Error(ctx, span, err, err.Error())
+			c.HandleAPIError(w, r, apierrors.NewErrInternal(err))
```

Comment:
i believe telemetry.Error also formats the error like you did, so you can just do
```suggestion
			err = telemetry.Error(ctx, span, err, "error getting token for API")
			c.HandleAPIError(w, r, apierrors.NewErrPassThroughToClient(err, http.StatusInternalServerError))
```

Reply:
It doesn't, that why I'm doing it this way :(

I think it should, but that change is more invasive, so I'll save it for later.